packages/near-api-js/test/providers.test.js

```diff
@@ -295,6 +295,41 @@ test('json rpc light client proof', async() => {
     await expect(provider.lightClientProof(lightClientRequest)).rejects.toThrow(/.+ block .+ is ahead of head block .+/);
 });
 
+test('json rpc get next light client block', withProvider(async (provider) => {
+    const stat = await provider.status();
+
+    // Get block in at least the last epoch (epoch duration 43,200 blocks on mainnet and testnet)
+    const height = stat.sync_info.latest_block_height;
+    const protocolConfig = await provider.experimental_protocolConfig({ finality: 'final' });
+
+    if (height < protocolConfig.epoch_length) {
+        // Not enough blocks to query one from last epoch, query current and assert that it returns
+        // empty. This may have a race condition where the query overlaps a new block being produced
+        let res = await provider.nextLightClientBlock({ last_block_hash: stat.sync_info.latest_block_hash });
+        if (res) {
+            // It's possible that the first epoch ended during this test (edge case),
+            // ensure that requesting with this new epoch will return the empty result
+            res = await provider.nextLightClientBlock({ last_block_hash: res.next_block_inner_hash });
+        }
+        expect(res).toStrictEqual({});
```

Comment:
I'm a little confused here, is the intention always to throw away the first result else fail the test?

Reply:
no, if the result returns `{}` (meaning there is not a newer block produced while the test is running), it doesn't send another request. This is just to ensure there isn't the race condition of a block being produced and indexed since the test started.

Definitely thinking about it more, there still is a race condition with this test, so maybe it's just better to remove checking that it's at the current head. I just liked the idea of verifying that it's at the head, which is why that code path is used.